### DISCUSSION TITLE
Does Coraza V3 support CRS stable version v3.3.4?

### DISCUSSION THREAD

**Discussion by bingfengfeifei** [corazawaf/coraza#817] [Q&A] [answered]
I was able to successfully load CRS v4 rules with Coraza V3, but encountered an error when trying to load CRS rules of version v3.3.4. The error message was "invalid WAF config from string: failed to parse string: failed to compile the directive 'secrule': invalid arguments, expected collection TX". Is this a normal phenomenon?

**Comment by M4tteoP** [comment, marked as answer]
Hey @bingfengfeifei, thanks for reaching out. I confirm that CRS v3.3.4 is not supported by Coraza v3. CRS v4 comes with a refactored way of generating the regexes, making them compatible with Go (RE2).
